File: fix-emoji.mjs
#!/usr/bin/env node
/**
 * Fix broken emoji encoding in breed/page.tsx
 */

import { readFileSync, writeFileSync } from 'fs';

// Mapping of broken encodings to correct emojis
const EMOJI_MAP = {
  'рџ§¬': '🧬',  // DNA
  'вЂў': '•',   // bullet point
  'вљЎ': '⚗️',  // test tube/flask
  'вњ…': '✅',  // check mark
  'рџ"—': '📗',  // green book
  'рџ'ё': '💸',  // money flying
  'вљҐ': '⚥',   // gender symbols
  'рџЋІ': '🎲',  // dice
  'вЏ±пёЏ': '⏱️',  // stopwatch
  'вљ пёЏ': '⚠️',  // warning
  'вљ°пёЏ': '⚰️',  // coffin
  'рџљ«': '🚫',  // prohibited
  'вЏі': '⏳',   // hourglass
  'рџ"'': '🔒',  // locked
  'OCTAAвЂ"WMON': 'OCTAA–WMON',  // em dash
};

function fixEmojiEncoding(filePath) {
  console.log(`Fixing emoji encoding in ${filePath}...`);
  
  // Read file with UTF-8 encoding
  let content = readFileSync(filePath, 'utf8');
  
  // Count replacements
  let totalReplacements = 0;
  
  // Replace broken encodings with correct emojis
  for (const [broken, correct] of Object.entries(EMOJI_MAP)) {
    const regex = new RegExp(broken.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'g');
    const matches = content.match(regex);
    const count = matches ? matches.length : 0;
    
    if (count > 0) {
      content = content.replace(regex, correct);
      totalReplacements += count;
      console.log(`  Replaced '${broken}' → '${correct}' (${count} times)`);
    }
  }
  
  // Write back to file
  writeFileSync(filePath, content, 'utf8');
  
  console.log(`✅ Done! Total replacements: ${totalReplacements}`);
  return totalReplacements;
}

// Run fix
const filePath = 'app/breed/page.tsx';
const total = fixEmojiEncoding(filePath);

if (total === 0) {
  console.log('⚠️ No broken emojis found. File may already be fixed.');
} else {
  console.log(`🎉 Fixed ${total} broken emojis!`);
}

File: fix-emoji.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""Fix broken emoji encoding in breed/page.tsx"""

import re

# Mapping of broken encodings to correct emojis
EMOJI_MAP = {
    'рџ§¬': '🧬',  # DNA
    'вЂў': '•',   # bullet point
    'вљЎ': '⚗️',  # test tube/flask
    'вњ…': '✅',  # check mark
    'рџ"—': '📗',  # green book
    'рџ'ё': '💸',  # money flying
    'вљҐ': '⚥',   # gender symbols
    'рџЋІ': '🎲',  # dice
    'вЏ±пёЏ': '⏱️',  # stopwatch
    'вљ пёЏ': '⚠️',  # warning
    'вљ°пёЏ': '⚰️',  # coffin
    'рџљ«': '🚫',  # prohibited
    'вЏі': '⏳',   # hourglass
    'рџ"'': '🔒',  # locked
    'OCTAAвЂ"WMON': 'OCTAA–WMON',  # em dash
}

def fix_emoji_encoding(file_path):
    """Fix broken emoji encoding in file"""
    print(f"Fixing emoji encoding in {file_path}...")
    
    # Read file with UTF-8 encoding
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Count replacements
    total_replacements = 0
    
    # Replace broken encodings with correct emojis
    for broken, correct in EMOJI_MAP.items():
        count = content.count(broken)
        if count > 0:
            content = content.replace(broken, correct)
            total_replacements += count
            print(f"  Replaced '{broken}' → '{correct}' ({count} times)")
    
    # Write back to file
    with open(file_path, 'w', encoding='utf-8', newline='\n') as f:
        f.write(content)
    
    print(f"✅ Done! Total replacements: {total_replacements}")
    return total_replacements

if __name__ == '__main__':
    file_path = 'app/breed/page.tsx'
    total = fix_emoji_encoding(file_path)
    
    if total == 0:
        print("⚠️ No broken emojis found. File may already be fixed.")
    else:
        print(f"🎉 Fixed {total} broken emojis!")
